### PR TITLE
[CTS] New KeyEventNotification/Traces & Update Tracker API

### LINKED ISSUE #422: [CTS] API Updated
https://confluence.tsi-dev.otc-service.com/display/DR/RM-3679-CTS-API

### PR DESCRIPTION
**Pull request by Aloento**
Move all API methods and related structures to a separate file.
Inline all `Extracts` and use the `extract` package instead.
Remove all Builders

tracker/List -> Get

New:
KeyEventNotification
traces/List

fix #422

### CODE REVIEW

File: openstack/cts/v1/tracker/results.go [hunk 1]
```diff
@@ -1,58 +1,49 @@
...
+
+	IsSupportTraceFilesEncryption bool   `json:"is_support_trace_files_encryption"`
+	CreateTime                    int64  `json:"create_time"`
+	StreamId                      string `json:"streamId"`
+	GroupId                       string `json:"groupId"`
+	IsSupportValidate             bool   `json:"is_support_validate"`
+	TrackerType                   string `json:"tracker_type"`
+	DomainId                      string `json:"domain_id"`
+	ProjectId                     string `json:"project_id"`
+	Id                            string `json:"id"`
```
Comment (anton-sidelnikov) [inline openstack/cts/v1/tracker/results.go:29]: is it possible to add doc here?

File: openstack/cts/v1/tracker/results.go [hunk 2]
```diff
@@ -1,58 +1,49 @@
...
+	Lts         Lts    `json:"lts"`
+
+	IsSupportTraceFilesEncryption bool   `json:"is_support_trace_files_encryption"`
+	CreateTime                    int64  `json:"create_time"`
+	StreamId                      string `json:"streamId"`
+	GroupId                       string `json:"groupId"`
+	IsSupportValidate             bool   `json:"is_support_validate"`
+	TrackerType                   string `json:"tracker_type"`
+	DomainId                      string `json:"domain_id"`
+	ProjectId                     string `json:"project_id"`
```
Comment (anton-sidelnikov) [inline openstack/cts/v1/tracker/results.go:28]: is it possible to add doc here?

**Review by anton-sidelnikov — CHANGES_REQUESTED** [review]

File: openstack/cts/v1/tracker/results.go (hunk 2 above)
Comment (Aloento) [reply openstack/cts/v1/tracker/results.go:28]: I couldn't find any documentation related to them, but they actually exist.

**Comment by Aloento** [comment]
@anton-sidelnikov @artem-lifshits please review this thank you

File: acceptance/openstack/cts/v1/trackers_test.go [hunk 3]
```diff
@@ -12,56 +14,46 @@ func TestTrackersLifecycle(t *testing.T) {
...
+			IsLtsEnabled: pointerto.Bool(true),
 		},
-	}
-	ctsTracker, err := tracker.Create(client, createOpts).Extract()
+	})
+
 	t.Cleanup(func() {
 		t.Logf("Attempting to delete CTSv1 Tracker: %s", ctsTracker.TrackerName)
-		err := tracker.Delete(client).ExtractErr()
+		err := tracker.Delete(client)
```
Comment (anton-sidelnikov) [inline acceptance/openstack/cts/v1/trackers_test.go]: this is really works? it should delete by name: DELETE /v1.0/{project_id}/tracker?tracker_name=system

File: openstack/cts/v1/tracker/Delete.go [hunk 4]
```diff
@@ -0,0 +1,9 @@
+package tracker
+
+import "github.com/opentelekomcloud/gophertelekomcloud"
+
+// Delete will permanently delete a particular tracker.
+func Delete(client *golangsdk.ServiceClient) (err error) {
```
Comment (anton-sidelnikov) [inline openstack/cts/v1/tracker/Delete.go]: why you not passing tracker name here?

**Review by anton-sidelnikov — CHANGES_REQUESTED** [review]
Please add tests for keyevent

**Review by anton-sidelnikov — APPROVED** [review]

**Comment by Aloento** [comment]
@artem-lifshits @Polina-Gubina

**Review by Polina-Gubina — APPROVED** [review]

**Comment by anton-sidelnikov** [comment]
regate

**Review by artem-lifshits — APPROVED** [review]

**Pull request merged**